provider/openstack/cinder.go

```diff
@@ -595,7 +595,7 @@ func cinderToJujuVolumeInfo(volume *cinder.Volume) storage.VolumeInfo {
 
 func detachVolume(instanceId, volumeId string, storageAdapter OpenstackStorage) error {
 	err := storageAdapter.DetachVolume(instanceId, volumeId)
-	if err != nil && !errors.IsNotFound(err) {
+	if err != nil && !(errors.IsNotFound(err) || gooseerrors.IsNotFound(err)) {
```

Comment:
This could be expressed as `err != nil && !errors.IsNotFound(err) && !gooseerrors.IsNotFound(err)`

Reply:
I went with Christian's suggestion of putting errors.IsNotFound as well as gooseerrors.IsNotFound into one provider method. It does make reading code easier.